packages/dds/tree/src/shared-tree/sharedTree.ts

```diff
@@ -330,6 +330,13 @@ export class SharedTree
 				this.commitEnricher.commitTransaction();
 			}
 		});
+		this.checkout.events.on("afterBatch", (newCommits) => {
```

Comment:
This is a little weird, but I think it's at least a step in the right direction. The commit enricher has a dependency on both the commit graph _and_ the index state (e.g. state of the forest at a certain point in time). So, it makes sense that it would respond to "afterBatch". I forwarded the new commits through the event for simplicity.

Reply:
But isn't afterBatch fired after the edits are applied to the forest? The commit enricher needs to told about them beforehand.

Reply:
Does it? That means that the previous code was relying on the fact that its event subscription to `beforeChange` (in EditManager) fired before the same subscription in TreeCheckout. That's not good - we don't guarantee that events fire in order of subscription (though they currently do in practice) and relying on that ordering is a very obfuscated way of embedding a sequencing dependency in the system.

What should we do then? Add "beforeBatch"? ☹️

Reply:
Yeah, sorry about that, I've been bad. There is currently no way for the user to experience an issue if the commit enricher is told about the change after the forest is updated. This is because there's no way to make an edit that would uncover the bug. This will stop being true soon though.

I would love for the forest to be updated on afterBatch or some other event that is guaranteed to fire after the commit enricher is notified.

Reply:
For now, I've added `beforeBatch` and used that for the commit enrichment. I think we can untangle it, probably, in a future PR. 

We can't have the forest be updated on `afterBatch` because the forest updating is what causes `afterBatch` to fire, by definition. But I'm all for revisiting all of these events and whether they are serving us best, and changing them if not.